pkgs/servers/nosql/eventstore/default.nix

```diff
@@ -1,80 +1,84 @@
-{ lib, stdenv
-, fetchFromGitHub
-, fetchurl
-, makeWrapper
+{ lib
+, git
 , dotnetCorePackages
+, glibcLocales
+, buildDotnetModule
+, fetchFromGitHub
+, bintools
+, stdenv
 , mono
-, Nuget
+, patchelf
 }:
 
-let
-
-  dotnet-sdk = dotnetCorePackages.sdk_5_0;
-
-  deps = import ./deps.nix { inherit fetchurl; };
-
-in
-
-stdenv.mkDerivation rec {
-
+buildDotnetModule rec {
   pname = "EventStore";
-  version = "5.0.8";
+  version = "21.10.5";
 
   src = fetchFromGitHub {
     owner = "EventStore";
     repo = "EventStore";
     rev = "oss-v${version}";
-    sha256 = "021m610gzmrp2drywl1q3y6xxpy4qayn580d855ag952z9s6w9nj";
+    sha256 = "sha256-uUDjTGCiQgXmvOUsujIA0JkGQGuw9U4zLKDP1WIFq1o=";
+    leaveDotGit = true;
   };
 
-  buildInputs = [
-    makeWrapper
-    dotnet-sdk
-    mono
-    Nuget
-  ];
+  # Fixes application reporting 0.0.0.0 as its version.
+  MINVERVERSIONOVERRIDE = version;
 
-  # that dependency seems to not be required for building, but pulls in libcurl which fails to be located.
-  # see: https://github.com/EventStore/EventStore/issues/1897
-  patchPhase = ''
-    for f in $(find . -iname "*.csproj"); do
-      sed -i '/Include="Microsoft.SourceLink.GitHub"/d' $f
-    done
-  '';
+  dotnet-sdk = dotnetCorePackages.sdk_5_0;
+  dotnet-runtime = dotnetCorePackages.aspnetcore_5_0;
+
+  nativeBuildInputs = [ git glibcLocales bintools ];
 
-  buildPhase = ''
-    export FrameworkPathOverride=${mono}/lib/mono/4.7.1-api
+  runtimeDeps = [ mono ];
 
-    # disable default-source so nuget does not try to download from online-repo
-    nuget sources Disable -Name "nuget.org"
-    # add all dependencies to a source called 'nixos'
-    for package in ${toString deps}; do
-      nuget add $package -Source nixos
+  nugetBinariesToPatch = lib.optionals stdenv.isLinux [
+    "grpc.tools/2.39.1/tools/linux_x64/protoc"
+    "grpc.tools/2.39.1/tools/linux_x64/grpc_csharp_plugin"
+  ];
+
+  postConfigure = ''
+    # Fixes execution of native protoc binaries during build
+    for binary in $nugetBinariesToPatch; do
```

Comment:
I'm not sure how common execution of native binaries from nuget packages is, but maybe this could be apart of `buildDotnetModule` if other packages need it as well. 

Note that this isn't blocking for this PR, I just wanted to point it out.

Reply:
Yeah, I had that idea also. I would also like to figure out some way to run native nuget binaries outside of `buildDotnetModule` (ie. when running `dotnet` in shell). This would for example fix AOT compilation (which last time I checked, fails.

I would prefer to address this in a separate PR however, since this one is big enough.

Reply:
Made an issue as a TODO - https://github.com/NixOS/nixpkgs/issues/181373

Reply:
> Yeah, I had that idea also. I would also like to figure out some way to run native nuget binaries outside of buildDotnetModule (ie. when running dotnet in shell). This would for example fix AOT compilation (which last time I checked, fails.

That would definitely be nice to have, thanks for creating the issue.

> I would prefer to address this in a separate PR however, since this one is big enough.

I entirely agree.